crates/ruff_python_parser/src/token_source.rs

```diff
@@ -89,26 +91,26 @@ impl<'src> TokenSource<'src> {
     /// token vector. It does add the trivia tokens to the token vector.
     fn do_bump(&mut self) {
         loop {
-            let next = self.lexer.next_token();
-            if next.is_trivia() {
-                if next.is_comment() {
-                    self.comments.push(next.range());
+            let kind = self.lexer.next_token();
+            if is_trivia(kind) {
+                if kind == TokenKind::Comment {
+                    self.comments.push(self.current_range());
                 }
-                self.tokens.push(next);
+                self.tokens.push(Token::new(kind, self.current_range()));
                 continue;
             }
             break;
         }
     }
 
-    /// Returns the next non-trivia token without adding it to any vector.
+    /// Returns the next non-trivia token without adding it to the token vector.
     fn next_non_trivia_token(&mut self) -> TokenKind {
         loop {
-            let next = self.lexer.next_token();
-            if next.is_trivia() {
+            let kind = self.lexer.next_token();
+            if is_trivia(kind) {
```

Comment:
There's already a `is_trivia` method on `TokenKind` which also considers indentation and logical newlines. I plan to look into resolving the differences.